### ISSUE TITLE
reorganization of user interface

### ISSUE THREAD

**Issue by rhodey** [WhisperSystems/Flock#33]
To resolve issue #30 a new action will need to be made available in the UI somehow so I figure this is a good time to discuss ideas for potential reworks of the user interface.

I would like to add a slide-out menu to the main activity (PreferencesActivity) similar to the slide-out menu present in the conversation list view of TextSecure. In that menu would be the following items:
1. "Import / Export"
2. "Change Password"
3. "Batch Delete" (or something like that...)

The import / export option would bring you to a FragmentActivity with two tabs up top, one reading "Import", the other "Export". Clicking on the tabs switches which fragment is loaded. From the import fragment you could do the following:
1. import contacts from accounts.
2. import contacts from storage. (eg SD card or internal storage)
3. import calendars from accounts.
4. import calendars from storage.

From the export fragment you could do the following:
1. export contacts to storage.
2. export calendars to storage.

"Batch Delete" is where I want to hide the "clear all contacts from Flock account" UI mentioned in #30.

The above changes would mean the "Import contacts" and "Import calendars" menu items could be dropped from the main UI. In addition to this the "My calendars" menu item and "Default calendar color" option should be replaced with some visual representation of a calendar indicating the number of calendars being synced and default calendar color. Clicking on this visual calendar representation will launch the "My calendars" UI. Same goes for the "My addressbooks" menu item (though this option is only visible to non-flock sync users).

As for the "SYNC" section of the main menu, I would like to drop the "Sync on content change" option entirely and replace the "Sync frequency" option with some visual representation of the option. 

The "ACCOUNT" section of the main menu will no longer contain "Change encryption password" because this is now inside the slide-out drawer. Then I want to move the "Unregister account" action into the "Manage subscription" UI and replace the "Manage subscription" main menu item with a clickable icon that displays the number of subscription days remaining.

Basically minimize the amount of text within the main menu UI by replacing standard menu items and settings with icons and other fancy jazz. Would greatly appreciate any feedback and suggestions :)

**Comment by phipolis** [comment]
One thing that's caught my imagination about Flock came from the release announcement on the OWS blog: "What's next? For calendars and contacts, Flock just works on Android. But we’re not stopping there. Encrypted backup of photos, videos, and desktop integration are all on the horizon." 

Obviously that's still a ways out! But, this reminds me of that. A valuable goal in reorganizing the UI might be to maintain a generic interface, with room to grow beyond just contacts and calendars.

For example, consider how Android's "Google" Acounts pane is structured: It shows the account name, sync status, and some general settings. Clicking on the account name brings up a list of specific items being synced. Each has a checkbox, and unchecking them will (purportedly) delete the records from the server. 

That obviously isn't enough for Flock, a 'sync' checkbox alone is too simplistic for the options you're discussing. Generally though, I think it makes sense to move the account and backup/restore/delete options to a separate place, leaving the main activity for a (more or less) concise way to list and control whatever synced services there are.

[Also, worth noting that TextSecure has now removed its slide out menu in favor of placing "Import/Export" and "Settings" into the three-dots menu.]
